wgengine/router/router_linux_test.go

```diff
@@ -595,3 +599,69 @@ func (o *fakeOS) output(args ...string) ([]byte, error) {
 	}
 	return []byte(strings.Join(ret, "\n")), nil
 }
+
+var tunTestNum int64
+
+func createTestTUN(t *testing.T) tun.Device {
+	const minimalMTU = 1280
+	tunName := fmt.Sprintf("tuntest%d", atomic.AddInt64(&tunTestNum, 1))
+	tun, err := tun.CreateTUN(tunName, minimalMTU)
+	if err != nil {
+		t.Fatalf("CreateTUN(%q): %v", tunName, err)
+	}
+	return tun
+}
+
+func TestDelRouteIdempotent(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("test requires root")
```

Comment:
technically I think CAP_NET_ADMIN is sufficient

Reply:
yeah. I wish the tools for working with Linux capabilities were easier. 

I could use `go test --exec=wrapper-that-uses-sudo-and-setcap` but then the wrapper is already 
using sudo so what's the point? (Yeah, less code runs as root.)  But still tedious. I wish there were a sudo-with-only-some-capabilities or something.